src/Microsoft.AspNet.Mvc.Core/FilterActionInvoker.cs

```diff
@@ -146,6 +152,149 @@ private IFilter[] GetFilters()
             }
         }
 
+        private async Task InvokeAllResourceFiltersAsync()
+        {
+            _cursor.SetStage(FilterStage.ResourceFilters);
+
+            var context = new ResourceExecutingContext(ActionContext, _filters);
+
+            context.InputFormatters = new List<IInputFormatter>(_inputFormatterProvider.InputFormatters);
+            context.InputFormatterSelector = _inputFormatterSelector;
```

Comment:
I don't think the selector should be changeable by the user (at least not via a ResourceFilter). 
```InputFormatterSelector``` does on the input side what ```ObjectResult``` does on the output side. It selects the input formatter by calling ```CanRead``` on input formatter.